server/auth.go

```diff
@@ -557,6 +578,54 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 		err  error
 		ao   bool // auth override
 	)
+
+	// Check if we have auth callouts enabled at the server level or in the bound account.
+	defer func() {
+		if juc == nil && opts.AuthCallout == nil {
+			return
+		}
+		// We have a juc define here, check account.
+		if juc != nil && !acc.hasExternalAuth() {
+			return
+		}
+
+		// We have auth callout set here.
+		var skip bool
+		if authorized {
+			// Check if we are on the list of auth_users.
+			userID := c.getRawAuthUser()
+			if juc != nil {
+				skip = acc.isExternalAuthUser(userID)
+			} else {
+				for _, u := range opts.AuthCallout.AuthUsers {
+					if userID == u {
+						skip = true
+						break
+					}
+				}
+			}
+		}
+		// If we are here we have an auth callout defined and we have failed auth so far
+		// so we will callout to our auth backend for processing.
+		if !skip {
+			authorized = s.processClientOrLeafCallout(c, opts)
+		}
+		// Check if we are and in the auth callout account, and if so add in deny publish permissions for the auth subject.
+		if authorized {
+			var authAccountName string
+			if juc == nil && opts.AuthCallout != nil {
+				authAccountName = opts.AuthCallout.Account
+			} else if juc != nil {
+				authAccountName = acc.Name
+			}
+			c.mu.Lock()
```

Comment:
It looks like the `if`/`else if` above isn't exhaustive, so if `authAccountName` isn't found for some reason, does it make sense to take out the lock and continue?

Reply:
The account connected to that for the static config test is checked on startup. This is just checking if we are in that account and making sure this client can not send requests on the server behalf.